iothub/device/samples/how to guides/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -256,70 +251,45 @@ private async void ConnectionStatusChangeHandlerAsync(ConnectionStatus status, C
-            {
-                _initSemaphore.Release();
+                _logger.LogDebug($"The desired property version cached on local is changing from {s_localDesiredPropertyVersion} to {serverDesiredPropertyVersion}.");
+                await HandleTwinUpdateNotificationsAsync(twinCollection, cancellationToken);
             }
         }
 
         private async Task HandleTwinUpdateNotificationsAsync(TwinCollection twinUpdateRequest, object userContext)
         {
-            var cancellationToken = (CancellationToken)userContext;
+            var reportedProperties = new TwinCollection();
 
-            if (!cancellationToken.IsCancellationRequested)
-            {
-                var reportedProperties = new TwinCollection();
+            _logger.LogInformation($"Twin property update requested: \n{twinUpdateRequest.ToJson()}");
 
-                _logger.LogInformation($"Twin property update requested: \n{twinUpdateRequest.ToJson()}");
-
-                // For the purpose of this sample, we'll blindly accept all twin property write requests.
-                foreach (KeyValuePair<string, object> desiredProperty in twinUpdateRequest)
-                {
-                    _logger.LogInformation($"Setting property {desiredProperty.Key} to {desiredProperty.Value}.");
-                    reportedProperties[desiredProperty.Key] = desiredProperty.Value;
-                }
+            // For the purpose of this sample, we'll blindly accept all twin property write requests.
+            foreach (KeyValuePair<string, object> desiredProperty in twinUpdateRequest)
+            {
+                _logger.LogInformation($"Setting property {desiredProperty.Key} to {desiredProperty.Value}.");
+                reportedProperties[desiredProperty.Key] = desiredProperty.Value;
+            }
 
-                s_localDesiredPropertyVersion = twinUpdateRequest.Version;
-                _logger.LogDebug($"The desired property version on local is currently {s_localDesiredPropertyVersion}.");
+            s_localDesiredPropertyVersion = twinUpdateRequest.Version;
+            _logger.LogDebug($"The desired property version on local is currently {s_localDesiredPropertyVersion}.");
 
+            try
+            {
                 // For the purpose of this sample, we'll blindly accept all twin property write requests.
-                await RetryOperationHelper.RetryTransientExceptionsAsync(
-                    operationName: "UpdateReportedProperties",
-                    asyncOperation: async () => await s_deviceClient.UpdateReportedPropertiesAsync(reportedProperties, cancellationToken),
-                    shouldExecuteOperation: () => IsDeviceConnected,
-                    logger: _logger,
-                    exceptionsToBeIgnored: _exceptionsToBeIgnored,
-                    cancellationToken: cancellationToken);
+                await s_deviceClient.UpdateReportedPropertiesAsync(reportedProperties, s_appCancellation.Token);
```

Comment:
Found an issue that the SDK does not protect against exceptions thrown by this callback (and probably others). Until that is dealt with, we should at least model not letting exceptions get through, at least the one we know can happen.